lib/cache-driver.js

```diff
@@ -137,10 +137,6 @@ function CacheDriver(callContext, authority, resource, clientId, cache, refreshF
  * @param  {QueryCallback} callback
  */
 CacheDriver.prototype._find = function(query, callback) {
-  var authorityQuery = {};
```

Comment:
I thought that the behavior we are after is to always look for a tenant specific token first.  If we can't find one of those then we attempt to us an MRRT to get a token for the other tenant.  If that doesn't work the we just fail.  I don't see that happening here.  You've removed the authority clause of the cache queries, but I don't see where it's been added back anywhere in more particular circumstances.  Did I miss it?